src/ncurses_ui.cc

```diff
@@ -661,12 +661,51 @@ Optional<Key> NCursesUI::get_next_key()
         const int new_c = wgetch(m_window);
         if (new_c == '[') // potential CSI
         {
-            const Codepoint csi_val = wgetch(m_window);
-            switch (csi_val)
+            const Codepoint c1 = wgetch(m_window);
+            switch (c1)
             {
                 case 'I': return {Key::FocusIn};
                 case 'O': return {Key::FocusOut};
-                default: break; // nothing
+                case '1':
+                {
+                  const Codepoint c2 = wgetch(m_window);
+                  if ( c2 != ';' )
+                  {
+                      ungetch(c2); ungetch(c1); break;
```

Comment:
I understand why the calls to `ungetch` are on the same line, but I don't think the `break` should be on that line as well. Applies to the other hunks as well.